### DISCUSSION TITLE
Set variable value inside the execution block

### DISCUSSION THREAD

**Discussion by rtscosta** [Huachao/vscode-restclient#1227] [Q&A]
I made a global variable named @transactionId. It should only be assigned a value if I run the request ### approve task! When I use ###, it sets a value for it, which is fine. However, I also set values for this variable in another part of the code. 

###
@transactionId = {{completTask.response.body.$.data.transactionId}}

### approve task
# @name approveTask
PATCH http://localhost:3000/api/v1/task/{{idTask}}/approve
Content-Type: application/json
Authorization: Bearer {{accessTokenDeTask}}

{
}

###
@transactionId = {{approveTask.response.body.$.data.transactionId}} //When we reach that line, the value is overwritten because I ran this method once before.

**Comment by MominRaza** [comment]
Related update: https://github.com/Huachao/vscode-restclient/pull/1422

This discussion is addressed by the new `@set` directive, which lets you capture response values (`response.headers.*` / `response.body.*`) into shared runtime variables and reuse them across `.http` files.
